File: src/services/notificationService.js
// Serviço unificado para notificações
class NotificationService {
  constructor() {
    this.apiUrl = process.env.VITE_API_URL || 'https://fhd-automacao-industrial-bq67.vercel.app'
  }

  // Enviar notificação para app mobile
  async sendMobileNotification(type, data) {
    try {
      const response = await fetch(`${this.apiUrl}/api/notify-mobile`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ type, data })
      })

      if (!response.ok) {
        throw new Error('Erro ao enviar notificação mobile')
      }

      const result = await response.json()
      
      // Abrir WhatsApp automaticamente
      if (result.notification?.whatsappUrl) {
        window.open(result.notification.whatsappUrl, '_blank')
      }

      return result
    } catch (error) {
      console.error('Erro no serviço de notificação:', error)
      throw error
    }
  }

  // Notificar boleto vencendo
  async notifyOverdueBill(billData) {
    return await this.sendMobileNotification('overdue_bill', billData)
  }

  // Notificar novo orçamento
  async notifyNewQuotation(quotationData) {
    return await this.sendMobileNotification('new_quotation', quotationData)
  }

  // Notificar divisão de lucro
  async notifyProfitSharing(profitData) {
    return await this.sendMobileNotification('profit_sharing', profitData)
  }

  // Testar notificação
  async testNotification() {
    const testData = {
      company_name: 'Empresa Teste',
      amount: '1.500,00',
      due_date: '2024-01-15'
    }
    
    return await this.notifyOverdueBill(testData)
  }
}

const notificationService = new NotificationService()
export default notificationService 

File: mobile/lib/services/notification_service.dart
import 'dart:convert';
import 'package:flutter/foundation.dart';
import 'package:http/http.dart' as http;
import 'package:flutter_local_notifications/flutter_local_notifications.dart';
import 'package:shared_preferences/shared_preferences.dart';

class NotificationService extends ChangeNotifier {
  final FlutterLocalNotificationsPlugin _notifications = FlutterLocalNotificationsPlugin();
  String _serverUrl = '';
  bool _isConnected = false;
  List<Map<String, dynamic>> _notificationsList = [];

  String get serverUrl => _serverUrl;
  bool get isConnected => _isConnected;
  List<Map<String, dynamic>> get notificationsList => _notificationsList;

  Future<void> initialize() async {
    // Carregar configurações salvas
    final prefs = await SharedPreferences.getInstance();
    _serverUrl = prefs.getString('server_url') ?? 'https://fhd-automacao-industrial-bq67.vercel.app';

    // Configurar notificações locais
    const androidSettings = AndroidInitializationSettings('@mipmap/ic_launcher');
    const iosSettings = DarwinInitializationSettings();
    
    const initSettings = InitializationSettings(
      android: androidSettings,
      iOS: iosSettings,
    );

    await _notifications.initialize(initSettings);

    // Solicitar permissões
    await _requestPermissions();
  }

  Future<void> _requestPermissions() async {
    final androidSettings = AndroidFlutterLocalNotificationsPlugin();
    await androidSettings.requestNotificationsPermission();
  }

  Future<void> setServerUrl(String url) async {
    _serverUrl = url;
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString('server_url', url);
    notifyListeners();
  }

  Future<bool> testConnection() async {
    try {
      final response = await http.get(Uri.parse('$_serverUrl/api/notify-mobile'));
      _isConnected = response.statusCode == 200;
      notifyListeners();
      return _isConnected;
    } catch (e) {
      _isConnected = false;
      notifyListeners();
      return false;
    }
  }

  Future<void> sendTestNotification() async {
    try {
      final response = await http.post(
        Uri.parse('$_serverUrl/api/notify-mobile'),
        headers: {'Content-Type': 'application/json'},
        body: json.encode({
          'type': 'overdue_bill',
          'data': {
            'company_name': 'Empresa Teste',
            'amount': '1.500,00',
            'due_date': '2024-01-15'
          }
        }),
      );

      if (response.statusCode == 200) {
        await _showLocalNotification(
          'Teste de Notificação',
          'Conexão com servidor funcionando!',
        );
        
        // Adicionar à lista
        _notificationsList.insert(0, {
          'title': 'Teste de Notificação',
          'message': 'Conexão com servidor funcionando!',
          'timestamp': DateTime.now().toIso8601String(),
          'type': 'test'
        });
        notifyListeners();
      }
    } catch (e) {
      await _showLocalNotification(
        'Erro de Conexão',
        'Não foi possível conectar com o servidor',
      );
    }
  }

  Future<void> _showLocalNotification(String title, String body) async {
    const androidDetails = AndroidNotificationDetails(
      'whatsapp_notifier_channel',
      'WhatsApp Notifier',
      channelDescription: 'Canal para notificações do WhatsApp Notifier',
      importance: Importance.high,
      priority: Priority.high,
    );

    const notificationDetails = NotificationDetails(
      android: androidDetails,
    );

    await _notifications.show(
      DateTime.now().millisecondsSinceEpoch.remainder(100000),
      title,
      body,
      notificationDetails,
    );
  }

  void clearNotifications() {
    _notificationsList.clear();
    notifyListeners();
  }
} 